src/ArrowFunctions.cpp

```diff
@@ -0,0 +1,151 @@
+#include "ArrowFunctions.h"
+
+#include <iostream>
+#include <arrow/api.h>
+#include <arrow/io/api.h>
+#include <parquet/arrow/reader.h>
+#include <parquet/arrow/writer.h>
+
+int cpp_getNumRows(const char* filename) {
+  std::shared_ptr<arrow::io::ReadableFile> infile;
+  PARQUET_ASSIGN_OR_THROW(
+      infile,
+      arrow::io::ReadableFile::Open(filename,
+                                    arrow::default_memory_pool()));
+
+  std::unique_ptr<parquet::arrow::FileReader> reader;
+  PARQUET_THROW_NOT_OK(
+    parquet::arrow::OpenFile(infile, arrow::default_memory_pool(), &reader));
+  return reader -> parquet_reader() -> metadata() -> num_rows();
+}
```

Comment:
Are readers & file handles reusable or do you have to construct new ones each time?

Starting from the filename is good because you can't really screw it up, but I wonder if we need to be worried about the overhead (probably not since we generally are going to read a file once and be done).

Also another question, I noticed the reader  & file objects don't get explicitly closed.  Does that happen automatically / is it something we need to worry about?

Reply:
The issue here is we can't pass a C++ object to C/Chapel, so there is no way for us to get the opened file back to Chapel to reuse.

Yes, the files are closed automaticall (see https://arrow.apache.org/docs/cpp/api/formats.html#parquet-reader)

Reply:
FWIW I don't expect there to be a noticeable overhead from not reusing. The actual file read/write should dominate the time.